watlow/util.py

```diff
@@ -120,21 +120,14 @@ async def _request(self, method, *args, **kwargs):
         """
         await self.connectTask
         async with self.lock:
-            if not self.client.connected:
-                raise TimeoutError("Not connected to Watlow gateway.")
-            future = getattr(self.client.protocol, method)(*args, **kwargs)
             try:
-                return await asyncio.wait_for(future, timeout=self.timeout)
-            except asyncio.TimeoutError as e:
-                if self.client.connected and hasattr(self, 'modbus'):
-                    # This came from reading through the pymodbus@python3 source
-                    # Problem was that the driver was not detecting disconnect
-                    self.client.protocol_lost_connection(self.modbus)
-                raise TimeoutError(e)
-            except pymodbus.exceptions.ConnectionException as e:
-                if self.client.connected and hasattr(self, 'modbus'):
-                    self.client.protocol_lost_connection(self.modbus)
-                raise ConnectionError(e)
+                if self.pymodbus32plus:
+                    future = getattr(self.client, method)
+                else:
+                    future = getattr(self.client.protocol, method)  # type: ignore
```

Comment:
Pedantic, but I'm used to seeing version-dependent logic at the top of the file right beneath imports.

Very much against the usual pattern of minimizing scope, but it's still a standard I'm used to seeing.

Don't know how best to do that here, but food for thought either way.

Reply:
I can't see a way to get the version-_dependent_ code up to the top of the file / module, because `self.client` won't be initialized by then.  Pretty sure it needs to stay here.


There is version-_detection_ code which could theoretically be moved to the top of the file.
https://github.com/numat/watlow/blob/ba1174c97ef69882194563e6679a7766f6d60de7/watlow/util.py#L44-L47

That could be used for the version dependent imports which use `try/except` currently.
https://github.com/numat/watlow/blob/ba1174c97ef69882194563e6679a7766f6d60de7/watlow/util.py#L8-L13

... but then I'd have to use globals instead of using attributes of `AsyncioModbusClient`.  I'm not sure that's any cleaner.

Thanks for the thought exercise. However... It Ain't Broke so I'm Not Gonna Fix It. :)